src/libs/actions/Report.js

```diff
@@ -660,7 +661,7 @@ function handleReportChanged(report) {
     // A report can be missing a name if a comment is received via pusher event
     // and the report does not yet exist in Onyx (eg. a new DM created with the logged in person)
     if (report.reportID && report.reportName === undefined) {
-        openReport(report.reportID);
+        reconnect(report.reportID);
```

Comment:
From looking at the code of `openReport`, I understand that this updates the last time you read the report, which makes sense if you were "opening" it.

Here instead, we are likely receiving a report comment in a pusher event and we don't have the report. If we do `openReport`, we wrongly mark it as read as if we had opened it. `reconnect` is better for this case because we don't update the report read mark.

It this correct?

I'm wondering why the bug doesn't happen when we just send a message instead of an IOU

Reply:
If the above is correct, I would suggest adding a comment on why we use `reconnect` and not `openReport`, mentioning the last read stuff.

Reply:
Yes, that's correct. 

This code only gets called when `report.reportName` is undefined. Which seems to be the case for IOUs/split bills that come from someone you've never chatted with. But not for regular chat messages. I'm not really sure why that is 🤔

Reply:
Added comment.

Reply:
thanks!, I'm trying to test in dev, but I'm having problems with the pusher events

Reply:
> But not for regular chat messages. I'm not really sure why that is 🤔

I think the action of opening a new chat creates it and pushes it to the other participants, even if you haven't sent a message yet.

![image](https://user-images.githubusercontent.com/87341702/218862106-0f20c2bf-bd26-4868-a471-e4a328a8d6af.png)

so by the time you send a message, it is likely that the other participants would already have the reports (unless they missed the those push events)